re-organzied port-agent and added argocd installation method (#762)
port-labs/port-docs: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md

---
sidebar_position: 1
---

# Helm

> Let's dive in to a walkthrough on how to install the Port execution agent in your Kubernetes cluster.

Reviewer: See comment in argocd.md line 7

:::info

> You can observe the helm chart with the full installation [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-agent).

Reviewer 2: ```suggestion
You can observe the helm chart and the available parameters [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-agent).
```

:::

:::note prerequisites

> - The [Helm](https://helm.sh) must be installed to use the chart. Please refer to

Reviewer: ```suggestion
- [Helm](https://helm.sh) must be installed to use the chart. Please refer to
```

>   the Helm's [documentation](https://helm.sh/docs) for further details on the installation;

Reviewer: ```suggestion
  the [Helm documentation](https://helm.sh/docs) for further details about the installation.
```

- The connection credentials to Kafka are provided to you by Port.
- If you want to trigger a GitLab Pipeline, you need to have a [GitLab trigger token](https://docs.gitlab.com/ee/ci/triggers/)

:::

1. Add Port's Helm repo by using the following command:

```bash showLineNumbers

> helm repo add port-labs https://port-labs.github.io/helm-charts

Reviewer 2: Let's add both the `add` and the `repo update` commands in the same code block. I don't want to rely on the user to remember that he installed in the past

```

> If you already added this repo earlier, run `helm repo update` to retrieve

Reviewer 2: We can remove this part completely after adding the `helm repo update` command to the above code block

the latest versions of the charts. You can then run `helm search repo port-labs` to see the charts.

2. Install the `port-agent` chart by using the following command:

> ## Installation

Reviewer: Move the "Installation" title up, it should be directly below the prerequisites like you did in argocd.md

:::note
Remember to replace the placeholders for `YOUR_ORG_ID`, `YOUR_KAFKA_CONSUMER_GROUP`, `YOUR_PORT_CLIENT_ID` and `YOUR_PORT_CLIENT_SECRET`.
:::

```bash showLineNumbers
helm install my-port-agent port-labs/port-agent \
    --create-namespace --namespace port-agent \
    --set env.normal.PORT_ORG_ID=YOUR_ORG_ID \
    --set env.normal.KAFKA_CONSUMER_GROUP_ID=YOUR_KAFKA_CONSUMER_GROUP \
    --set env.secret.PORT_CLIENT_ID=YOUR_PORT_CLIENT_ID \
    --set env.secret.PORT_CLIENT_SECRET=YOUR_PORT_CLIENT_SECRET
```

## Next Steps

> - [Usage](/create-self-service-experiences/self-service-actions-deep-dive/self-service-actions-deep-dive.md) guide Set up a webhook.

Reviewer: See comment at the end of argocd.md
